LibreHardwareMonitorLib/Hardware/SMBios.cs

```diff
@@ -648,6 +648,16 @@ internal BiosInformation(byte[] data, IList<string> strings) : base(data, string
             int.TryParse(parts[1], out int day) &&
             int.TryParse(parts[2], out int year))
         {
+            if (day > 31)
+            {
+                (year, day) = (day, year);
+            }
+
+            if (month > 31)
```

Comment:
This doesn't seem right, months are already checked below, and there are only 12 months in a year.